mws/resource_mws_private_access_settings.go

```diff
@@ -2,110 +2,77 @@ package mws
-func (a PrivateAccessSettingsAPI) Create(pas *PrivateAccessSettings) error {
-	pasAPIPath := fmt.Sprintf("/accounts/%s/private-access-settings", pas.AccountID)
-	return a.client.Post(a.context, pasAPIPath, pas, &pas)
-}
-
-// Read returns the PAS object along with metadata and any additional errors
-func (a PrivateAccessSettingsAPI) Read(mwsAcctID, pasID string) (PrivateAccessSettings, error) {
-	var pas PrivateAccessSettings
-	pasAPIPath := fmt.Sprintf("/accounts/%s/private-access-settings/%s", mwsAcctID, pasID)
-	err := a.client.Get(a.context, pasAPIPath, nil, &pas)
-	return pas, err
-}
-
-func (a PrivateAccessSettingsAPI) Update(pas *PrivateAccessSettings) error {
-	pasAPIPath := fmt.Sprintf("/accounts/%s/private-access-settings/%s", pas.AccountID, pas.PasID)
-	return a.client.Put(a.context, pasAPIPath, pas)
-}
-
-// Delete deletes the PAS object given a pas id
-func (a PrivateAccessSettingsAPI) Delete(mwsAcctID, pasID string) error {
-	pasAPIPath := fmt.Sprintf("/accounts/%s/private-access-settings/%s", mwsAcctID, pasID)
-	if err := a.client.Delete(a.context, pasAPIPath, nil); err != nil {
-		return err
-	}
-	return nil
-}
-
-// List lists all the available PAS objects in the mws account
-func (a PrivateAccessSettingsAPI) List(mwsAcctID string) ([]PrivateAccessSettings, error) {
-	var pasList []PrivateAccessSettings
-	pasAPIPath := fmt.Sprintf("/accounts/%s/private-access-settings", mwsAcctID)
-	err := a.client.Get(a.context, pasAPIPath, nil, &pasList)
-	return pasList, err
-}
-
 func ResourceMwsPrivateAccessSettings() *schema.Resource {
-	s := common.StructToSchema(PrivateAccessSettings{}, func(s map[string]*schema.Schema) map[string]*schema.Schema {
+	s := common.StructToSchema(provisioning.PrivateAccessSettings{}, func(s map[string]*schema.Schema) map[string]*schema.Schema {
 		// nolint
 		s["private_access_settings_name"].ValidateFunc = validation.StringLenBetween(4, 256)
+		common.SetRequired(s["private_access_settings_name"])
+		common.SetRequired(s["region"])
+
 		s["private_access_level"].ValidateFunc = validation.StringInSlice([]string{"ACCOUNT", "ENDPOINT"}, true)
+		common.SetDefault(s["private_access_level"], "ACCOUNT")
+
+		s["private_access_settings_id"].Computed = true
+
+		common.AddAccountIdField(s)
 		return s
 	})
 	p := common.NewPairSeparatedID("account_id", "private_access_settings_id", "/")
 	return common.Resource{
 		Schema: s,
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var pas PrivateAccessSettings
+			a, err := c.AccountClientWithAccountIdFromConfig(d)
+			if err != nil {
+				return err
+			}
+			var pas provisioning.UpsertPrivateAccessSettingsRequest
 			common.DataToStructPointer(d, s, &pas)
 			common.SetForceSendFields(&pas, d, []string{"public_access_enabled"})
-			if err := NewPrivateAccessSettingsAPI(ctx, c).Create(&pas); err != nil {
+			res, err := a.PrivateAccess.Create(ctx, provisioning.UpsertPrivateAccessSettingsRequest(pas))
```

Comment:
Why do we create a new `provisioning.UpsertPrivateAccessSettingsRequest` ?

Reply:
Good catch. This doesn't make a new one but just does a type conversion to itself. Removing this now.